src/sagemaker/session.py

```diff
@@ -126,6 +140,9 @@ def _initialize(self, boto_session, sagemaker_client, sagemaker_runtime_client):
 
         prepend_user_agent(self.sagemaker_runtime_client)
 
+        self._default_bucket = None
+        self._desired_default_bucket_name = default_bucket
```

Comment:
l. 143 is already being done in l. 105. I would just move l. 144 to l. 106. You also then wouldn't have to duplicate the logic in the local mode test object.

and maybe `_default_bucket_name_override`?

Reply:
+1. Fixed!